pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,1002 @@
+                               end_doy=end_doy)
+        estr = 'Day of year (end_doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    def test_basic_instrument_load_yr_no_end_doy(self):
+        """Ensure `end_doy` required if `end_yr` present."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, self.ref_doy,
+                               self.ref_time.year)
+        estr = 'Both end_yr and end_doy must be set'
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("input", [{'yr': 2009, 'doy': 1,
+                                        'date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'end_date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'fname': 'dummy_str.nofile'},
+                                       {'yr': 2009, 'doy': 1,
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile',
+                                        'end_yr': 2009, 'end_doy': 1}])
+    def test_basic_instrument_load_mixed_inputs(self, input):
+        """Ensure mixed load inputs raise ValueError.
+
+        parameters
+        ----------
+        input : dict
+            Dictionary of keywords and arguments to produce an error when
+            instrument is loaded.
+
+        """
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(**input)
+        estr = 'An inconsistent set of inputs have been'
+        assert str(err).find(estr) >= 0
+        return
+
+    def test_basic_instrument_load_no_input(self):
+        """Test that `.load()` loads all data."""
+
+        self.testInst.load()
+        assert (self.testInst.index[0] == self.testInst.files.start_date)
+        assert (self.testInst.index[-1] >= self.testInst.files.stop_date)
+        assert (self.testInst.index[-1] <= self.testInst.files.stop_date
+                + dt.timedelta(days=1))
+        return
+
+    @pytest.mark.parametrize('load_in,verr',
+                             [('fname', 'have multi_file_day and load by file'),
+                              (None, 'is not supported with multi_file_day')])
+    def test_instrument_load_errors_with_multifile(self, load_in, verr):
+        """Ensure load by filename raises ValueError with `multi_file_day` as True.
```

Comment:
Since this handles multiple cases, how about this:
```suggestion
        """Ensure improper usage of load with `multi_file_day` as True raises ValueError.
```

Reply:
tagging @rstoneback since this is outside the other comment thread